core/lib/types/src/l1/mod.rs

```diff
@@ -265,159 +264,116 @@ impl L1Tx {
+        assert_eq!(req.transaction.nonce, req.tx_id); // serial id from decoded from transaction bytes should be equal to one from event
+        assert_eq!(req.transaction.max_priority_fee_per_gas, U256::zero());
+        assert_eq!(req.transaction.paymaster, U256::zero());
+        assert_eq!(req.transaction.hash(), H256::from_slice(&req.tx_hash));
+        let factory_deps_hashes: Vec<_> = req
+            .factory_deps
+            .iter()
+            .map(|b| h256_to_u256(hash_bytecode(b)))
+            .collect();
+        assert_eq!(req.transaction.factory_deps, factory_deps_hashes);
+        for item in &req.transaction.reserved[2..] {
             assert_eq!(item, &U256::zero());
         }
-
-        let calldata = transaction.remove(0).into_bytes().unwrap();
-
-        let signature = transaction.remove(0).into_bytes().unwrap();
-        assert_eq!(signature.len(), 0);
-
+        assert!(req.transaction.signature.is_empty());
         // TODO (SMA-1621): check that `reservedDynamic` are constructed correctly.
-        let _factory_deps_hashes = transaction.remove(0).into_array().unwrap();
-        let _paymaster_input = transaction.remove(0).into_bytes().unwrap();
-        let _reserved_dynamic = transaction.remove(0).into_bytes().unwrap();
-
-        // Decoding metadata
-
-        // Finally, decode the factory dependencies
-        let factory_deps = match dec_ev.remove(0) {
-            Token::Array(factory_deps) => factory_deps,
-            _ => unreachable!(),
-        };
-
-        let factory_deps = factory_deps
-            .into_iter()
-            .map(|token| token.into_bytes().unwrap())
-            .collect::<Vec<_>>();
+        assert!(req.transaction.paymaster_input.is_empty());
+        assert!(req.transaction.reserved_dynamic.is_empty());
 
         let common_data = L1TxCommonData {
-            serial_id,
-            canonical_tx_hash,
-            sender,
+            serial_id: PriorityOpId(req.transaction.nonce.try_into().unwrap()),
+            canonical_tx_hash: H256::from_slice(&req.tx_hash),
+            sender: u256_to_account_address(&req.transaction.from),
             layer_2_tip_fee: U256::zero(),
-            to_mint,
-            refund_recipient,
+            to_mint: req.transaction.reserved[0],
+            refund_recipient: u256_to_account_address(&req.transaction.reserved[1]),
             full_fee: U256::zero(),
-            gas_limit,
-            max_fee_per_gas,
-            gas_per_pubdata_limit,
+            gas_limit: req.transaction.gas_limit,
+            max_fee_per_gas: req.transaction.max_fee_per_gas,
+            gas_per_pubdata_limit: req.transaction.gas_per_pubdata_byte_limit,
             op_processing_type: OpProcessingType::Common,
             priority_queue_type: PriorityQueueType::Deque,
             // DEPRECATED.
-            // TODO (PLA-962): start setting it to 0 for all new transactions.
-            eth_block,
+            eth_block: 0,
```

Comment:
i guess eth_watch still relies on these numbers being non-zero. if the output of this method is not intended for eth watcher, then better to add a comment about it

Reply:
Yes, that's why it is being set in the TryFrom<Log> below.